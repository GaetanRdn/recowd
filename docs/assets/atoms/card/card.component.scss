@mixin rc-card($config) {
  $typography: map-get($config, typography);
  $colors: map-get($config, colors);
  $primary: map-get($colors, primary);

  rc-card {
    display: inline-flex;
    flex-flow: column;
    height: fit-content;
    min-width: fit-content;
    border-radius: 0.8rem;
    box-shadow: rgba(0, 0, 0, 12%) 0 3px 5px, rgba(0, 0, 0, 24%) 0 3px 5px;

    @each $key in color, background-color {
      #{$key}: map-get($primary, $key);
    }
  }

  .rc-card-header {
    border-top-left-radius: 0.8rem;
    border-top-right-radius: 0.8rem;
  }

  .rc-card-header:not(:empty) {
    display: grid;
    grid-template-columns: auto 1fr;
    column-gap: 1rem;
    row-gap: 0.4rem;
    align-items: center;
    padding: 1.6rem 1.6rem 0 1.6rem;

    .rc-avatar {
      grid-row: 1 / span 2;
    }

    .rc-card-title {
      grid-row: 1 / span 1;
    }

    .rc-card-subtitle {
      grid-row: 2 / span 1;
    }
  }

  .rc-card-content {
    display: block;
    padding: 1.6rem;

    $bodyRegular: map-get($typography, body-regular);

    @each $key in font-size, font-family, font-weight, line-height {
      #{$key}: map-get($bodyRegular, $key);
    }
  }

  .rc-card-title {
    $heading: map-get($typography, heading);

    @each $key in font-size, font-family, font-weight, line-height {
      #{$key}: map-get($heading, $key);
    }

    display: block;
  }

  .rc-card-subtitle {
    $subheading: map-get($typography, body-regular);

    @each $key in font-size, font-family, font-weight, line-height {
      #{$key}: map-get($subheading, $key);
    }

    display: block;
    color: darken(map-get($primary, color), 30%);
  }
}
