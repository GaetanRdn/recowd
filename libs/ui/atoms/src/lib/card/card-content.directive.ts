import { Directive } from '@angular/core';

@Directive({
  selector: '[rcCardContent], rc-card-content',
  host: { class: 'rc-card-content ' },
})
export class CardContentDirective {}
