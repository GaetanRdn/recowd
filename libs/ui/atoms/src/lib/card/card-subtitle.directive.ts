import { Directive } from '@angular/core';

@Directive({
  selector: '[rcCardSubtitle], rc-card-subtitle',
  host: { class: 'rc-card-subtitle ' },
})
export class CardSubtitleDirective {}
