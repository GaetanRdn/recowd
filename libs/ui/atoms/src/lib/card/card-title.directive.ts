import { Directive } from '@angular/core';

@Directive({
  selector: '[rcCardTitle], rc-card-title',
  host: { class: 'rc-card-title ' },
})
export class CardTitleDirective {}
