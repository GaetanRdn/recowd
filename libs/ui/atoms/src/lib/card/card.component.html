<div class="rc-card-header">
  <ng-content select="[rcAvatar]"></ng-content>
  <ng-content select="rc-card-title, [rcCardTitle]"></ng-content>
  <ng-content select="rc-card-subtitle, [rcCardSubtitle]"></ng-content>
</div>
<ng-content select="rc-card-content, [rcCardContent]"></ng-content>
