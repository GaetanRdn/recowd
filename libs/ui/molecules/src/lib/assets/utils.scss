@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
