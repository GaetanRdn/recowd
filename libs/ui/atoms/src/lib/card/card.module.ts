import { NgModule } from '@angular/core';
import { CardComponent } from './card.component';
import { CardContentDirective } from './card-content.directive';
import { CardTitleDirective } from './card-title.directive';
import { CardSubtitleDirective } from './card-subtitle.directive';
import { CommonModule } from '@angular/common';

@NgModule({
  declarations: [
    CardComponent,
    CardContentDirective,
    CardTitleDirective,
    CardSubtitleDirective,
  ],
  exports: [
    CardComponent,
    CardContentDirective,
    CardTitleDirective,
    CardSubtitleDirective,
  ],
  imports: [CommonModule],
})
export class CardModule {}
