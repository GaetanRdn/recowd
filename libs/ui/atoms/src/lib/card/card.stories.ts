import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { CardComponent } from './card.component';
import { CardModule } from './card.module';
import { AvatarDirective, AvatarModule } from '../avatar/avatar.directive';
import { BADGE } from '@geometricpanda/storybook-addon-badges';
import { CardContentDirective } from './card-content.directive';
import { CardTitleDirective } from './card-title.directive';
import { CardSubtitleDirective } from './card-subtitle.directive';

export default {
  title: 'atoms/card',
  component: CardComponent,
  subcomponents: {
    CardContentDirective,
    CardTitleDirective,
    CardSubtitleDirective,
    AvatarDirective,
  },
  parameters: {
    badges: [BADGE.STABLE],
    jest: 'card.component',
  },
  decorators: [
    moduleMetadata({
      imports: [CardModule, AvatarModule],
    }),
  ],
} as Meta<CardComponent>;

const template: Story<CardComponent> = () => ({
  template: `<rc-card>
<rc-card-content>Simple card</rc-card-content>
</rc-card>`,
});

export const BasicCard = template.bind({});

const templateWithTitle: Story<CardComponent> = () => ({
  template: `<rc-card>
<rc-card-title>Card Title</rc-card-title>
<rc-card-content>Lorem ipsum dolor sit amet, consectetur adipiscing elit. In sit amet hendrerit nisi, a bibendum ex. Donec dignissim risus quis mauris semper suscipit. In hac habitasse platea dictumst. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Etiam congue consequat diam, in feugiat quam hendrerit eget. Donec ornare non felis vitae hendrerit</rc-card-content>
</rc-card>`,
});

export const CardWithTitle = templateWithTitle.bind({});

const templateWithSubTitle: Story<CardComponent> = () => ({
  template: `<rc-card>
<rc-card-title>Card Title</rc-card-title>
<rc-card-subtitle>Card subtitle</rc-card-subtitle>
<rc-card-content>Lorem ipsum dolor sit amet, consectetur adipiscing elit. In sit amet hendrerit nisi, a bibendum ex. Donec dignissim risus quis mauris semper suscipit. In hac habitasse platea dictumst. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Etiam congue consequat diam, in feugiat quam hendrerit eget. Donec ornare non felis vitae hendrerit</rc-card-content>
</rc-card>`,
});

export const CardWithSubtitle = templateWithSubTitle.bind({});

const templateForFullCard: Story<CardComponent> = () => ({
  template: `<rc-card>
<img rcAvatar src="assets/venom.png" alt="avatar"/>
<rc-card-title>Card Title</rc-card-title>
<rc-card-subtitle>Card subtitle</rc-card-subtitle>
<rc-card-content>Lorem ipsum dolor sit amet, consectetur adipiscing elit. In sit amet hendrerit nisi, a bibendum ex. Donec dignissim risus quis mauris semper suscipit. In hac habitasse platea dictumst. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Etiam congue consequat diam, in feugiat quam hendrerit eget. Donec ornare non felis vitae hendrerit</rc-card-content>
</rc-card>`,
});

export const FullCard = templateForFullCard.bind({});
