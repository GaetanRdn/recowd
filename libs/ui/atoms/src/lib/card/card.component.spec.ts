import { Component, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { TestBed, waitForAsync } from '@angular/core/testing';
import { CardModule } from './card.module';
import { TemplateLookup } from '@recowd/test/utils';

describe('CardComponent', () => {
  beforeEach(waitForAsync(() =>
    TestBed.configureTestingModule({
      declarations: [
        SimpleCardHostComponent,
        CardWithTitleHostComponent,
        CardWithSubTitleHostComponent,
        CardWithAvatarHostComponent,
      ],
      imports: [CardModule],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents()));

  it('should create simple card', () => {
    const templateLookup: TemplateLookup<SimpleCardHostComponent> =
      new TemplateLookup(SimpleCardHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });

  it('should create a card with title', () => {
    const templateLookup: TemplateLookup<CardWithTitleHostComponent> =
      new TemplateLookup(CardWithTitleHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });

  it('should create a card with subtitle', () => {
    const templateLookup: TemplateLookup<CardWithSubTitleHostComponent> =
      new TemplateLookup(CardWithSubTitleHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });

  it('should create a card with an avatar', () => {
    const templateLookup: TemplateLookup<CardWithAvatarHostComponent> =
      new TemplateLookup(CardWithAvatarHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-card>
    <rc-card-content>Simple card</rc-card-content>
  </rc-card>`,
})
export class SimpleCardHostComponent {}

@Component({
  template: ` <rc-card>
    <rc-card-title>Card title</rc-card-title>
    <rc-card-content>Simple card</rc-card-content>
  </rc-card>`,
})
export class CardWithTitleHostComponent {}

@Component({
  template: ` <rc-card>
    <rc-card-title>Card title</rc-card-title>
    <rc-card-subtitle>Card subtitle</rc-card-subtitle>
    <rc-card-content>Simple card</rc-card-content>
  </rc-card>`,
})
export class CardWithSubTitleHostComponent {}

@Component({
  template: ` <rc-card>
    <rc-material-icon rcAvatar>home</rc-material-icon>
    <rc-card-title>Card title</rc-card-title>
    <rc-card-subtitle>Card subtitle</rc-card-subtitle>
    <rc-card-content>Simple card</rc-card-content>
  </rc-card>`,
})
export class CardWithAvatarHostComponent {}
