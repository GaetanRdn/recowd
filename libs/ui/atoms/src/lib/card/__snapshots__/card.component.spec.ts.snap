// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent should create a card with an avatar 1`] = `
<rc-card>
  <div
    class="rc-card-header"
  >
    <rc-material-icon
      rcavatar=""
    >
      home
    </rc-material-icon>
    <rc-card-title
      class="rc-card-title"
    >
      Card title
    </rc-card-title>
    <rc-card-subtitle
      class="rc-card-subtitle"
    >
      Card subtitle
    </rc-card-subtitle>
  </div>
  <rc-card-content
    class="rc-card-content"
  >
    Simple card
  </rc-card-content>
</rc-card>
`;

exports[`CardComponent should create a card with subtitle 1`] = `
<rc-card>
  <div
    class="rc-card-header"
  >
    <rc-card-title
      class="rc-card-title"
    >
      Card title
    </rc-card-title>
    <rc-card-subtitle
      class="rc-card-subtitle"
    >
      Card subtitle
    </rc-card-subtitle>
  </div>
  <rc-card-content
    class="rc-card-content"
  >
    Simple card
  </rc-card-content>
</rc-card>
`;

exports[`CardComponent should create a card with title 1`] = `
<rc-card>
  <div
    class="rc-card-header"
  >
    <rc-card-title
      class="rc-card-title"
    >
      Card title
    </rc-card-title>
  </div>
  <rc-card-content
    class="rc-card-content"
  >
    Simple card
  </rc-card-content>
</rc-card>
`;

exports[`CardComponent should create simple card 1`] = `
<rc-card>
  <div
    class="rc-card-header"
  />
  <rc-card-content
    class="rc-card-content"
  >
    Simple card
  </rc-card-content>
</rc-card>
`;
